pkg/profile/artifact.go

```diff
@@ -7,54 +7,181 @@ import (
 	"time"
 
 	kustomizev1 "github.com/fluxcd/kustomize-controller/api/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
 	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
 	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
 )
 
+// Status contains the status of the artifacts
+type Status struct {
+	ResourcesExist     bool
+	NotReadyConditions []metav1.Condition
+}
+
 // CreateArtifacts creates and inserts objects to the cluster to deploy the
 // profile as a HelmRelease.
 func (p *Profile) CreateArtifacts(ctx context.Context) error {
-	if err := p.createGitRepository(ctx); err != nil {
+	gitRes, helmRes, kustomizeRes, err := p.makeArtifacts()
+	if err != nil {
+		return err
+	}
+
+	p.log.Info("creating GitRepository", "resource", gitRes.ObjectMeta.Name)
+	if err := p.client.Create(ctx, gitRes); err != nil {
 		return fmt.Errorf("failed to create GitRepository resource: %w", err)
 	}
 
-	switch kind := p.definition.Spec.Artifacts[0].Kind; kind {
-	case "HelmChart":
-		if err := p.createHelmRelease(ctx); err != nil {
+	if helmRes != nil {
+		p.log.Info("creating HelmRelease", "resource", helmRes.ObjectMeta.Name)
+		if err := p.client.Create(ctx, helmRes); err != nil {
 			return fmt.Errorf("failed to create HelmRelease resource: %w", err)
 		}
-	case "Kustomize":
-		if err := p.createKustomization(ctx); err != nil {
+	} else {
```

Comment:
meh not really a fan of `else`s

Reply:
This code looks complex, I suspect we could simplify it with [`%T`](https://golang.org/pkg/fmt/)

Reply:
Temporary, once multiple artifacts is added this will change